SimG4Core/Application/python/g4SimHits_cfi.py

```diff
@@ -367,8 +368,6 @@
         EminTrack  = cms.double(0.01),
         PutHistory = cms.bool(False),
         DoFineCalo = cms.bool(False),
-        EminFineTrack = cms.double(10000.0),
```

Comment:
today may be energy thresholds are used not in an optimal way but why they are removed ? history of low-energy particles is a pure overhead. I would keep energy limits in one or another form.

Reply:
Due to Geant's LIFO processing it's hard to prune low energy tracks during the SIM step. You never know whether a daughter track can still create a hit in a fine detector, and if you pruned the low energy parent track too early you are left with orphan hits. These will be low energy hits for sure, but for the sake of perfect accountability I'd prefer to save all of them during the SIM step. The additional memory and disk usage is relatively small.

Reply:
@tklijnsma, have check CPU and memory for Run-2 ?

Reply:
I did checks for a few events, and memory increased by a few percent, but nothing dramatic. CPU seemed unchanged.

Reply:
Majority of particles in showers are low-energy e+-, gamma (below 10 MeV). In hadronic showers also neutrons. They should not be counted as a separate MC truth object but simply add energies to hits produced by more energetic tracks.

Reply:
I agree but which energetic parent exactly? It's very hard to determine this correctly at the SIM track processing step, because not all tracks have been processed yet. It is much easier to do at a later stage when the full decay tree and all hits are available. The problem is illustrated in slide 16 here: https://indico.cern.ch/event/959375/contributions/4038791/attachments/2113127/3554832/0930_hgcaldpgtruth_v3.pdf .